kubespawner/spawner.py

```diff
@@ -1486,44 +1528,95 @@ def _validate_image_pull_secrets(self, proposal):
         - `description`: Optional description of this profile displayed to the user.
         - `kubespawner_override`: a dictionary with overrides to apply to the KubeSpawner
           settings. Each value can be either the final value to change or a callable that
-          take the `KubeSpawner` instance as parameter and return the final value.
+          take the `KubeSpawner` instance as parameter and return the final value. This can
+          be further overridden by 'options'
+        - 'options': A dictionary of sub-options that allow users to further customize the
+          selected profile. Items should have a unique key representing the customization,
+          and the value is a dictionary with the following keys:
+          - 'display_name': Name used to identify this particular option
+          - 'choices': A dictionary containing list of choices for the user to choose from
+            to set the value for this particular option. The key is an identifier for this
+            choice, and the value is a dictionary with the following possible keys:
+            - 'display_name': Human readable display name for this choice.
```

Comment:
We have several nested dictionaries: profile, profile option, and profile option choice.

We specify `display_name`, `default`, and `kubespawner_override` in multiple places within these nested dictionaries.

I think this could be the cause of confusion by how similar everything looks even though it is quite logical.

I can live with this as it is and I can't come up with another config schema that I like more but I want to highlight this in case someone comes up with an idea about this.